### ISSUE TITLE
API Calls Failing Due to TypeError (0.29.0)

### ISSUE THREAD

**Issue by abominox** [DataDog/datadogpy#393]
Hello,

Since the release of version `0.29.0` two hours ago, I've been encountering the following error for multiple API calls:

```
Traceback (most recent call last):
  File "test.py", line 14, in <module>
    print (api.Monitor.get_all())
  File "/usr/local/lib/python3.7/site-packages/datadog/api/monitors.py", line 57, in get_all
    return super(Monitor, cls).get_all(**params)
  File "/usr/local/lib/python3.7/site-packages/datadog/api/resources.py", line 181, in get_all
    return APIClient.submit('GET', cls._resource_name, api_version, **params)
  File "/usr/local/lib/python3.7/site-packages/datadog/api/api_client.py", line 161, in submit
    response_obj['response_headers'] = response_headers
TypeError: list indices must be integers or slices, not str
```

Currently using Python version 3.7.2. So far, I have encountered this when using `api.Monitor.get_all()` and `api.Downtime.get_all()`. When I revert to `0.28.0`, I no longer see these errors for these API calls.

**Comment by apgmckay** [comment]
We have experienced the same error `TypeError: list indices must be integers, not str` issue when making calls to the `api.Monitor.get_all()` Python version 2.7.5.

**Comment by ross-bragg** [comment]
I'm also having this issue using Python 3.5. The issue appears to come from PR #378 - the response object appears to be decoded as a list and the response headers are added as though it's a dict.

**Comment by nmuesch** [comment]
Hey @abominox, @apgmckay and @ross-bragg I've just released version 0.29.1 of the datadogpy library that should resolve this issue. Thanks for the detailed issue report and the investigation. Let me know if you continue to experience any issues here.
